nuxeo-coldstorage/src/test/java/org/nuxeo/coldstorage/operations/TestColdStorageRendition.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * (C) Copyright 2021 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Abdoul BA<aba@nuxeo.com>
+ */
+
+package org.nuxeo.coldstorage.operations;
+
+import static javax.servlet.http.HttpServletResponse.SC_INTERNAL_SERVER_ERROR;
+import static javax.servlet.http.HttpServletResponse.SC_NOT_FOUND;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.io.FilenameUtils;
+import org.junit.Test;
+import org.nuxeo.coldstorage.ColdStorageConstants;
+import org.nuxeo.coldstorage.S3ColdStorageFeature;
+import org.nuxeo.common.utils.FileUtils;
+import org.nuxeo.ecm.automation.OperationException;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.Blobs;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.platform.rendition.Rendition;
+import org.nuxeo.ecm.platform.rendition.service.RenditionService;
+import org.nuxeo.ecm.platform.video.VideoConstants;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+
+/**
+ * @since 10.10
+ */
+@Features(S3ColdStorageFeature.class)
+public class TestColdStorageRendition extends AbstractTestColdStorageOperation {
+
+    @Test
+    public void shouldSelectDefaultRendition() throws IOException, OperationException {
+        DocumentModel document = createFileDocument(session, true);
+
+        transactionalFeature.nextTransaction();
+
+        document.refresh();
```

Comment:
`session.save()` would probably be enough here instead of these 2 lines, as you're not expecting any asynchronous process on the created File document.

Reply:
indeed, 

but the rendition will happen after, I think we need to have `transactionalFeature.nextTransaction();` at elast one time.